src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,47 @@
 package mate.academy.spring.config;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService,
+                          PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
-                .authorizeRequests()
+                .authorizeRequests().antMatchers("/register", "/inject").permitAll()
+                .antMatchers(HttpMethod.POST, "/cinema-halls").hasAuthority("ADMIN")
+                .antMatchers(HttpMethod.GET, "/cinema-halls").hasAnyAuthority("ADMIN", "USER")
+                .antMatchers(HttpMethod.GET, "/movies").hasAnyAuthority("ADMIN", "USER")
+                .antMatchers(HttpMethod.POST, "/movies").hasAuthority("ADMIN")
+                .antMatchers(HttpMethod.GET, "/movie-sessions/available")
+                .hasAnyAuthority("ADMIN", "USER")
+                .antMatchers(HttpMethod.POST, "/movie-sessions").hasAuthority("ADMIN")
+                .antMatchers(HttpMethod.PUT, "/movie-sessions").hasAuthority("ADMIN")
+                .antMatchers(HttpMethod.DELETE, "/movie-sessions").hasAuthority("ADMIN")
```

Comment:
```suggestion
                .antMatchers(HttpMethod.PUT, "/movie-sessions/**").hasAuthority("ADMIN")
                .antMatchers(HttpMethod.DELETE, "/movie-sessions/**").hasAuthority("ADMIN")
```
have you tested your app?